src/mpid/ch4/src/ch4r_request.h

```diff
@@ -128,4 +128,86 @@ static inline int MPIDI_anysource_matched(MPIR_Request * rreq, int caller, int *
     return mpi_errno;
 }
 
+#ifndef MPIDI_CH4_DIRECT_NETMOD
+/* mpi-layer request handle is connected to one of the request (the one with shm),
+ * so we can't simply free one at matching time. Both need be completed and freed
+ * at the same time.
+ *
+ * At matching time, we cancel one of them (but not freed).
+ * At completion time, we complete and free both.
+ */
+static inline int MPIDI_anysrc_try_cancel(MPIR_Request * rreq, int *is_cancelled)
```

Comment:
How about `bool *is_cancelled`?

Reply:
Also we are not cancelling `rreq`, are we? Perhaps the function should be called `MPIDI_anysrc_try_cancel_partner`?

Reply:
> How about `bool *is_cancelled`?

Maybe. My hesitation is most of our code base use int for bool, so having bool sporadically makes the code base appear inconsistent. I would like have a separate PR to massively update the bool type (if that is preferred).

Reply:
> Also we are not cancelling `rreq`, are we? Perhaps the function should be called `MPIDI_anysrc_try_cancel_partner`?

When cancel the partner fails, it is essentially cancelling `rreq` instead.

Reply:
> When cancel the partner fails, it is essentially cancelling `rreq` instead.

But that's up to the caller of this function, right? This functions is returning true/false if the partner request was successfully cancelled.

Reply:
It updates the `CANCEL_BIT` on `rreq`. 

Anyway, I think I agree with the name-change.

Reply:
Ah, now I see it. I had assumed it was happening outside this function.

Reply:
The function is more of a convenience wrap rather than a clean function with a single purpose. However, the `is_cancel` refers to whether the partner is cancelled, so as far as user interface goes, the name change makes sense.

Reply:
The functionality is mixed in (unfortunately) to avoid incur additional overhead.